artifacts/reader.py

```diff
@@ -4,17 +4,58 @@
 import abc
 import glob
 import os
+import yaml
 
 from artifacts import artifact
 from artifacts import definitions
 from artifacts import errors
 
-import yaml
-
 
-class ArtifactsReader(object):
+class BaseArtifactsReader(object):
   """Class that implements the artifacts reader interface."""
 
+  LABELS = None
+  SUPPORTED_OS = None
+
+  @property
+  def supported_os(self):
```

Comment:
Yes.  The purpose of the Artificer ForensicArtifacts server that I am building is to have an API to fetch/store/edit artifacts.  This tweak lets me add new OS/Types/Labels etc internally on the server and allow the reader to validate artifacts using them them as being correct.

Additionally on the Writer side it will eventually let me export artifacts from an artifact server using an arbitrary set of labels.  So if I  have labels I want for internal use, but aren't relevant to the broader set I can do proper validation using the additional labels, but when exporting to add back to a repo like this I can filter out unsupported values.